provider/pkg/provider/helm_release.go

```diff
@@ -324,6 +322,7 @@ func (r *helmReleaseProvider) Check(ctx context.Context, req *pulumirpc.CheckReq
 	urn := resource.URN(req.GetUrn())
 	label := fmt.Sprintf("Provider[%s].Check(%s)", r.name, urn)
 
+	failures := []*pulumirpc.CheckFailure{}
```

Comment:
nit: prefer simple declaration (`var failures []*pulumirpc.CheckFailure`) if we don't know the size beforehand.